js/ad_manager_vast.js

```diff
@@ -483,6 +546,85 @@ OO.Ads.manager(function(_, $) {
     this.initialize = function(amc) {
       this.amc = amc;
       this.amc.addPlayerListener(this.amc.EVENTS.INITIAL_PLAY_REQUESTED, _.bind(this.initialPlay, this));
+      this.amc.addPlayerListener(this.amc.EVENTS.FULLSCREEN_CHANGED, _.bind(_onFullscreenChanged, this));
+      this.amc.addPlayerListener(this.amc.EVENTS.SIZE_CHANGED, _onSizeChanged);
+    };
+
+    /**
+     * Callback when the media file is loaded. Once is loaded we can initialize the ad
+     * @private
+     */
+    var _initializeAd = _.bind(function() {
+      var eventName,
+          environmentVariables,
+          viewMode,
+          creativeData = {};
+
+      currentAd.data = currentAd.ad.data;
+
+      if (typeof iframe.contentWindow.getVPAIDAd !== 'function') {
+        OO.log('VPaid: Required function getVPAIDAd() is not defined.');
+        return;
+      }
+      currentAd.vpaidAd = iframe.contentWindow.getVPAIDAd();
+
+      // Subscribe to ad unit events
+      for (eventName in VPAID_EVENTS) {
+        currentAd.vpaidAd.subscribe(_.bind(_onAdEvent, this, VPAID_EVENTS[eventName]),
+            VPAID_EVENTS[eventName], this);
+      }
+
+      this._slot = _createUniqueElement();
+      this._videoSlot = this.amc.ui.adVideoElement[0];
+
+      environmentVariables = _.extend({
+        slot: this._slot,
+        videoSlot: this._videoSlot,
+        videoSlotCanAutoPlay: true
+      }, this.environmentVariables);
+
+      this._properties = {
+        adWidth: this._slot.offsetWidth,
+        adHeight: this._slot.offsetHeight,
+        adDesiredBitrate: 600,
+      };
+
+      viewMode = _getFsState() ? 'fullscreen' : 'normal';
+      creativeData = {
+        AdParameters: currentAd.ad.adParams
+      };
+
+      this.initAd(this._properties['adWidth'], this._properties['adHeight'], viewMode,
+                  this._properties['adDesiredBitrate'], environmentVariables, creativeData);
+    }, this);
+
+    /**
+     * Initializes the ad by sending the data to the ad unit.
+     * @public
+     * @method VPaid#initAd
```

Comment:
The string before the # should be the class name. In this case, it should be Vast and not VPaid. If this function is used for VPAID only, please make note of it in the jsdoc and function name if necessary.